### PR TITLE
Ubiquity refactoring

### PR DESCRIPTION
**Pull request by midoblgsm**
Changing interface to use request objects
Adding hostname to attach/detach requests
Changing constants to use camel case style

---
This change is [Reviewable](https://reviewable.io/reviews/ibm/ubiquity/31)

### CODE REVIEW

**Comment by midoblgsm** [comment]
Needs to check with k8s (@mohamed) and docker plugins (@aswarke)

File: resources/resources.go [hunk 1]
```diff
@@ -1,15 +1,9 @@
...
-	"encoding/json"
-	"fmt"
-	"net/http"
-)
-
 const (
-	SPECTRUM_SCALE     Backend = "spectrum-scale"
-	SPECTRUM_SCALE_NFS Backend = "spectrum-scale-nfs"
-	SOFTLAYER_NFS      Backend = "softlayer-nfs"
+	SpectrumScale    Backend = "spectrum-scale"
```
Comment (aswarke) [inline resources/resources.go]: Any reason for converting constants to camel case ?. Ideally we'd like to distinguish between constants and regular variables

Comment (aswarke) [inline resources/resources.go]: I would be in favor of upper case for constants

**Comment by midoblgsm** [comment]
The convention in golang is to use CamelCase even for constants

**Pull request merged**